crates/ruff_linter/src/linter.rs

```diff
@@ -260,6 +260,26 @@ pub fn check_path(
         }
     }
 
+    // Update fix applicability to account for overrides
+    if !settings.extend_safe_fixes.is_empty() || !settings.extend_unsafe_fixes.is_empty() {
+        for diagnostic in diagnostics.iter_mut() {
+            if let Some(fix) = &diagnostic.fix {
+                // Check unsafe before safe so if someone puts a rule in both we are conservative
+                if settings
+                    .extend_unsafe_fixes
+                    .contains(diagnostic.kind.rule())
+                    && fix.applicability().is_always()
+                {
+                    diagnostic.set_fix(fix.clone().with_applicability(Applicability::Sometimes))
```

Comment:
Does this clone matter? Is there a better way to do this?

Reply:
This would work:
```rust
            if let Some(fix) = diagnostic.fix.take() {
                // Check unsafe before safe so if someone puts a rule in both we are conservative
                if settings
                    .extend_unsafe_fixes
                    .contains(diagnostic.kind.rule())
                    && fix.applicability().is_always()
                {
                    diagnostic.set_fix(fix.with_applicability(Applicability::Sometimes));
                } else if settings.extend_safe_fixes.contains(diagnostic.kind.rule())
                    && fix.applicability().is_sometimes()
                {
                    diagnostic.set_fix(fix.with_applicability(Applicability::Always));
                } else {
                    diagnostic.set_fix(fix);
                }
            }
```
There's no `Fix::set_applicability(&mut self, ...)` (only a `mut self` method), so usual the `if let Some(fix) = &mut diagnostic.fix` trick doesn't work here

Reply:
Thanks!

Should I have made `set_applicability` take `&mut self`?